homeassistant/components/hvv_departures/sensor.py

```diff
@@ -0,0 +1,214 @@
+
+from aiohttp import ClientConnectorError
+from pygti.exceptions import InvalidAuth
+from pygti.gti import GTI, Auth
+
+from homeassistant.const import (
+    ATTR_ATTRIBUTION,
+    ATTR_ID,
+    CONF_HOST,
+    CONF_PASSWORD,
+    CONF_USERNAME,
+    DEVICE_CLASS_TIMESTAMP,
+)
+from homeassistant.helpers import aiohttp_client
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+
+from .const import ATTRIBUTION, DOMAIN, MANUFACTURER
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(minutes=1)
+MAX_LIST = 20
+MAX_TIME_OFFSET = 360
+ICON = "mdi:bus"
+UNIT_OF_MEASUREMENT = "min"
+
+ATTR_DEPARTURE = "departure"
+ATTR_LINE = "line"
+ATTR_ORIGIN = "origin"
+ATTR_DIRECTION = "direction"
+ATTR_TYPE = "type"
+ATTR_DELAY = "delay"
+ATTR_NEXT = "next"
+
+PARALLEL_UPDATES = 0
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+    """Set up the sensor platform."""
+
+    pass
+
+
+async def async_setup_entry(hass, config_entry, async_add_devices):
+    """Set up the sensor platform."""
+
+    session = aiohttp_client.async_get_clientsession(hass)
+
+    sensor = HVVDepartureSensor(hass, config_entry, session)
+    async_add_devices([sensor], True)
+
+
+class HVVDepartureSensor(Entity):
+    """HVVDepartureSensor class."""
+
+    def __init__(self, hass, config_entry, session):
+        """Initialize."""
+        self.hass = hass
+        self.config_entry = config_entry
+        self.station_name = self.config_entry.data["station"]["name"]
+        self.attr = {ATTR_ATTRIBUTION: ATTRIBUTION}
+        self._available = False
+        self._state = None
+        self._name = f"Departures at {self.station_name}"
+        self._last_error = None
+
+        self.gti = GTI(
+            Auth(
+                session,
+                self.config_entry.data[CONF_USERNAME],
+                self.config_entry.data[CONF_PASSWORD],
+                self.config_entry.data[CONF_HOST],
+            )
+        )
+
+    @Throttle(MIN_TIME_BETWEEN_UPDATES)
+    async def async_update(self, **kwargs):
+        """Update the sensor."""
+
+        departure_time = datetime.now() + timedelta(
+            minutes=self.config_entry.options.get("offset", 0)
+        )
+
+        payload = {
+            "station": self.config_entry.data["station"],
+            "time": {
+                "date": departure_time.strftime("%d.%m.%Y"),
+                "time": departure_time.strftime("%H:%M"),
+            },
+            "maxList": MAX_LIST,
+            "maxTimeOffset": MAX_TIME_OFFSET,
+            "useRealtime": self.config_entry.options.get("realtime", False),
+        }
+
+        if "filter" in self.config_entry.options:
+            payload.update({"filter": self.config_entry.options["filter"]})
+
+        try:
+            data = await self.gti.departureList(payload)
+
+            if data["returnCode"] == "OK" and len(data.get("departures", [])) > 0:
```

Comment:
You can simplify it by using the default return value of .get which is None and None is a falsy value so you can use it like this:
```suggestion
            if data["returnCode"] == "OK" and data.get("departures"):
```